src/porepy/numerics/fv/mpfa.py

```diff
@@ -266,11 +266,60 @@ def discretize(self, sd: pp.Grid, data: dict) -> None:
             loc_bnd: pp.BoundaryCondition = self._bc_for_subgrid(
                 active_bound, sub_sd, l2g_faces, active_grid
             )
+
+            # Eta can either be a scalar or a vector. If a vector valued eta is passed,
+            # it will have a length equal to the number of subfaces in the entire grid.
+            # If the grid is partitioned into subgrids, we need to adjust the length of
+            # eta to match the subfaces of the subgrid.
+            if isinstance(eta, np.ndarray):
```

Comment:
To me, it seems this functionality should be moved to fvutils. This is particularly the case if the corresponding code in mpsa.py is more or less equal (I have not done a careful comparison).